packages/mermaid/src/diagrams/sequence/sequenceRenderer.ts

```diff
@@ -425,7 +425,14 @@ const drawMessage = async function (diagram, msgModel, lineStartY: number, diagO
     }
   } else {
     line = diagram.append('line');
-    line.attr('x1', startx);
+    let adjustedStartx = startx;
+    if (
+      type === diagObj.db.LINETYPE.BIDIRECTIONAL_DOTTED ||
+      type === diagObj.db.LINETYPE.BIDIRECTIONAL_SOLID
+    ) {
+      startx < stopx ? (adjustedStartx += 3) : (adjustedStartx -= 3);
+    }
+    line.attr('x1', adjustedStartx);
```

Comment:
The startx and stopx values are calculated in the `buildMessageModel` function. 
Adding this adjustment here breaks the Separation of concern, as it makes the `drawMessage` function responsible for the adjustment, where it should only have drawn as per the dimensions calculated in `buildMessageModel`.

Reply:
```diff
diff --git a/packages/mermaid/src/diagrams/sequence/sequenceRenderer.ts b/packages/mermaid/src/diagrams/sequence/sequenceRenderer.ts
index 6bb3c654e..525358d90 100644
--- a/packages/mermaid/src/diagrams/sequence/sequenceRenderer.ts
+++ b/packages/mermaid/src/diagrams/sequence/sequenceRenderer.ts
@@ -425,14 +425,7 @@ const drawMessage = async function (diagram, msgModel, lineStartY: number, diagO
     }
   } else {
     line = diagram.append('line');
-    let adjustedStartx = startx;
-    if (
-      type === diagObj.db.LINETYPE.BIDIRECTIONAL_DOTTED ||
-      type === diagObj.db.LINETYPE.BIDIRECTIONAL_SOLID
-    ) {
-      startx < stopx ? (adjustedStartx += 3) : (adjustedStartx -= 3);
-    }
-    line.attr('x1', adjustedStartx);
+    line.attr('x1', startx);
     line.attr('y1', lineStartY);
     line.attr('x2', stopx);
     line.attr('y2', lineStartY);
@@ -1449,7 +1442,7 @@ const buildMessageModel = function (msg, actors, diagObj) {
   const [fromLeft, fromRight] = activationBounds(msg.from, actors);
   const [toLeft, toRight] = activationBounds(msg.to, actors);
   const isArrowToRight = fromLeft <= toLeft;
-  const startx = isArrowToRight ? fromRight : fromLeft;
+  let startx = isArrowToRight ? fromRight : fromLeft;
   let stopx = isArrowToRight ? toLeft : toRight;
 
   // As the line width is considered, the left and right values will be off by 2.
@@ -1488,6 +1481,14 @@ const buildMessageModel = function (msg, actors, diagObj) {
     if (![diagObj.db.LINETYPE.SOLID_OPEN, diagObj.db.LINETYPE.DOTTED_OPEN].includes(msg.type)) {
       stopx += adjustValue(3);
     }
+
+    if (
+      [diagObj.db.LINETYPE.BIDIRECTIONAL_SOLID, diagObj.db.LINETYPE.BIDIRECTIONAL_DOTTED].includes(
+        msg.type
+      )
+    ) {
+      startx -= adjustValue(3);
+    }
   }
 
   const allBounds = [fromLeft, fromRight, toLeft, toRight];
```

Reply:
Thanks for pointing out this function, I made the updates @sidharthv96